sunpy/io/_asdf.py

```diff
@@ -0,0 +1,109 @@
+from pathlib import Path
+from collections import OrderedDict
+
+import numpy as np
+
+import asdf
+
+from sunpy.io._header import FileHeader
+
+__all__ = ["write", "read", "get_header", "get_keys_name"]
+
+def write(fname, data, header, **kwargs):
+    """
+    Take ``(data, header)`` pairs and save it to an ASDF file.
+
+    Parameters
+    ----------
+    fname : `str`
+        File name, with extension.
+    data : `numpy.ndarray`
+        N-dimensional data array.
+    header : `dict`
+        A header dictionary.
+    """
+    map_name = Path(fname)
+    map_name = map_name.name
+    ty = data.dtype
+    data = data.astype(ty)
```

Comment:
i got a error while loading a asdf_file stating insufficient memory blocks so i tried this one and now I think it was just an error in local machine so I think this can be removed

Reply:
I removed this in https://github.com/Sauravroy34/sunpy/pull/1 and didn't see this issue. Is there a minimal example that when used with https://github.com/Sauravroy34/sunpy/pull/1 produces the error?